src/table/body-cell/td-element.tsx

```diff
@@ -20,55 +22,63 @@ export interface TableTdElementProps {
   stripedRows?: boolean;
   hasSelection?: boolean;
   hasFooter?: boolean;
-  isVisualRefresh?: boolean;
+  stickyColumnProperties: StickyColumnProperties;
+  tdRef?: React.Ref<HTMLTableCellElement>;
 }
 
-export const TableTdElement = React.forwardRef<HTMLTableCellElement, TableTdElementProps>(
-  (
-    {
-      className,
-      style,
-      children,
-      wrapLines,
-      isFirstRow,
-      isLastRow,
-      isSelected,
-      isNextSelected,
-      isPrevSelected,
-      nativeAttributes,
-      onClick,
-      isEvenRow,
-      stripedRows,
-      isVisualRefresh,
-      hasSelection,
-      hasFooter,
-    },
-    ref
-  ) => {
-    return (
-      <td
-        style={style}
-        className={clsx(
-          className,
-          styles['body-cell'],
-          wrapLines && styles['body-cell-wrap'],
-          isFirstRow && styles['body-cell-first-row'],
-          isLastRow && styles['body-cell-last-row'],
-          isSelected && styles['body-cell-selected'],
-          isNextSelected && styles['body-cell-next-selected'],
-          isPrevSelected && styles['body-cell-prev-selected'],
-          !isEvenRow && stripedRows && styles['body-cell-shaded'],
-          stripedRows && styles['has-striped-rows'],
-          isVisualRefresh && styles['is-visual-refresh'],
-          hasSelection && styles['has-selection'],
-          hasFooter && styles['has-footer']
-        )}
-        onClick={onClick}
-        ref={ref}
-        {...nativeAttributes}
-      >
-        {children}
-      </td>
-    );
-  }
-);
+export function TableTdElement({
+  className,
+  style,
+  children,
+  wrapLines,
+  isFirstRow,
+  isLastRow,
+  isSelected,
+  isNextSelected,
+  isPrevSelected,
+  nativeAttributes,
+  onClick,
+  isEvenRow,
+  stripedRows,
+  hasSelection,
+  hasFooter,
+  stickyColumnProperties,
+  tdRef,
+}: TableTdElementProps) {
+  const isVisualRefresh = useVisualRefresh();
+  const { stickyStyles, isSticky, isLastStickyLeft, isLastStickyRight } = stickyColumnProperties;
+  const { isStuckToTheLeft, isStuckToTheRight } = useStickyState(isLastStickyLeft, isLastStickyRight);
```

Comment:
The table cells are not memoized. Any state update causes the entire cell and its content to re-render. Now, with sticky columns, these updates can happen on-scroll, which is very often. I think we need to ensure somehow that every time the callback fires it only updates the `td` tag styles and nothing else. This can be achieved with memoization or, maybe, with imperative style alteration:

```
const callback = (stickyState) => {
   ref.current.classList.toggle(...)
   ref.current.style...
}

same styles need to be applied the React-way during the cell normal render
```

Uploading Screen Recording 2023-04-20 at 11.21.39.mov…